### ISSUE TITLE
perf(compute): advertise FusedAttn in vulkanBackend Caps and eliminate PrefillBatch CPU fallback

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12181]
# perf(compute): advertise FusedAttn in vulkanBackend Caps and eliminate PrefillBatch CPU fallback

```routing
lane: internal/compute
paths: internal/compute/vulkan.go, internal/compute/vulkan_ops.go
expected_steps: 4
```

**Lane:** internal/compute  
**Work Unit:** leaf  
**Generation:** gen/now  
**Priority:** P0  

## Parent context
AMD Strix Halo (gfx1151 / RDNA 3.5, 40 Compute Units, 128GB LPDDR5X-7500 unified memory) HAL capability advertising and GPU-native batched prefill execution in `anthony-chaudhary/fak`.

## Current state
`vulkanBackend.Caps()` in `internal/compute/vulkan.go:317` currently returns `Caps{DeviceMemory: true, UploadDtype: v.haveQ8, CapacityProbe: v.totalMem > 0, HostCapacityProbe: hostKnown, BatchedPrefill: true}` with `FusedAttn: false` (omitted). More critically, `PrefillBatch` in `internal/compute/vulkan_ops.go:101` falls back directly to the CPU reference implementation (`ref, ok := Default().(BatchedPrefillBackend); ref.PrefillBatch(args)`). On AMD Strix Halo, prompt prefill triggers this CPU fallback, pulling tensors from VRAM to host RAM and evaluating multi-token prefill on CPU cores, causing catastrophic prefill degradation down to single-digit tok/s.

## Why this is next
The presence of `Caps.FusedAttn` signals to the model execution loop that attention can be lowered to fused GPU kernels rather than unfused decomposed operations. Simultaneously, routing `PrefillBatch` through native Vulkan GPU pipelines rather than falling back to CPU reference is essential to keep all prefill tensor operations resident on the gfx1151 GPU.

## Working spine
Advertise `FusedAttn: true` in `internal/compute/vulkan.go:317` when Vulkan attention pipelines are compiled and available, and implement GPU-native `PrefillBatch` execution in `internal/compute/vulkan_ops.go` invoking Vulkan batched matmuls and attention without falling back to CPU reference, verified by `TestVulkanAttentionApprox` and `TestVulkanPrefillBatch`.

## Value
- Centrality: Core
- P1: advanced - advertises `Caps.FusedAttn=true` in `vulkanBackend.Caps()`, enabling model loops to dispatch fused GPU attention kernels.
- P2: advanced - eliminates CPU reference fallback in `PrefillBatch`, keeping multi-token prompt prefill entirely resident on the gfx1151 GPU.
- P3: preserved - preserves full numerical parity and exact token output equivalence with CPU reference batched prefill.
- P4: preserved - confined to `internal/compute` without leaking private types or modifying public interfaces.

## Core through-line
Upgrade Vulkan backend capability advertising and native batched prefill:
1. In `internal/compute/vulkan.go`, update `Caps()` to set `FusedAttn: true` when the device attention pipeline is initialized.
2. In `internal/compute/vulkan_ops.go`, eliminate the CPU reference fallback in `PrefillBatch(args PrefillBatchArgs)`.
3. Implement native Vulkan GPU dispatch for `PrefillBatch`:
   - Iterate or batch input token activations through device-resident embedding tables.
   - Dispatch GPU batched matrix multiplications (`q4k_matmul`, `q8_matmul`, or `matmul`) across layers.
   - Dispatch fused GPU attention operations over device KV-cache buffers.
   - Return `PrefillBatchResult` with device-resident logits and output states without transferring intermediate layers to host RAM.

## In scope
- Updating `vulkanBackend.Caps()` to advertise `FusedAttn: true`.
- Replacing CPU reference fallback in `PrefillBatch` with native Vulkan compute pipeline dispatches.
- Ensuring intermediate layer activations remain strictly device-resident in VRAM throughout multi-token prefill.
- Validating batched prefill outputs against single-token serial execution and CPU reference.

## Out of scope
- Modifying CUDA or Metal backend implementations.
- Modifying speculative decoding drafts or NPU offload channels.
- Host-side KV cache eviction or compression policies.

## Gold-plating boundary
Strictly excluded from this work unit:
- Dynamic JIT compilation of bespoke shader variants per matrix shape.
- Multi-device Vulkan instance multiplexing or distributed pipeline parallelism.
- Register pressure auto-tuning heuristics or vendor-specific proprietary assembly injections outside standard Vulkan SPIR-V extensions.

## Verifiable Witness
```bash
go test -tags vulkan -v ./internal/compute -run "TestVulkanAttentionApprox|TestVulkanPrefillBatch" -count=1
```

## Witness
```bash
go test -tags vulkan -v ./internal/compute -run "TestVulkanAttentionApprox|TestVulkanPrefillBatch" -count=1
```

### Non-Forgeable Witness
```bash
go test -tags vulkan -v ./internal/compute -run "TestVulkanAttentionApprox|TestVulkanPrefillBatch" -count=1
```

### File:Line Seam
- `internal/compute/vulkan.go:317`
- `internal/compute/vulkan_ops.go:101`

### Blast Radius & Affected Lanes
- **Primary Lane:** `internal/compute`
- **Unaffected Lanes:** `platform/*`, `tools/*`, `cmd/*` remain untouched. Main trunk remains green.

### Quarantined Fallback Mechanism
If native GPU `PrefillBatch` encounters an unhandled op or tensor format, it safely falls back to single-token serial Vulkan execution rather than crashing or stalling the serving engine.

## Likely files
- `internal/compute/vulkan.go`
- `internal/compute/vulkan_ops.go`

## Scoped Acceptance Criteria
- [ ] 1. [SW-VERIFIED] `internal/compute/vulkan.go` advertises `Caps.FusedAttn = true` in `vulkanBackend.Caps()` when attention pipelines are active.
- [ ] 2. [SW-VERIFIED] `internal/compute/vulkan_ops.go` implements GPU-native `PrefillBatch` without falling back to CPU reference.
- [ ] 3. [SW-VERIFIED] `TestVulkanAttentionApprox` and `TestVulkanPrefillBatch` pass with exact argmax agreement and cosine similarity >= 0.995.
- [ ] 4. [HW-WITNESSED] Physical appliance hardware execution on AMD Strix Halo (gfx1151) achieves >= 350 tok/s prefill throughput with 0 bytes transferred to CPU RAM during prefill layers.

## Acceptance criteria
- [ ] 1. [SW-VERIFIED] `internal/compute/vulkan.go` advertises `Caps.FusedAttn = true` in `vulkanBackend.Caps()` when attention pipelines are active.
- [ ] 2. [SW-VERIFIED] `internal/compute/vulkan_ops.go` implements GPU-native `PrefillBatch` without falling back to CPU reference.
- [ ] 3. [SW-VERIFIED] `TestVulkanAttentionApprox` and `TestVulkanPrefillBatch` pass with exact argmax agreement and cosine similarity >= 0.995.
- [ ] 4. [HW-WITNESSED] Physical appliance hardware execution on AMD Strix Halo (gfx1151) achieves >= 350 tok/s prefill throughput with 0 bytes transferred to CPU RAM during prefill layers.

## Done condition
`vulkanBackend.Caps()` advertises `FusedAttn: true`, `PrefillBatch` runs natively on GPU without CPU fallback, tests pass with cosine similarity >= 0.995, and physical appliance hardware execution on AMD Strix Halo achieves >= 350 tok/s prefill throughput with 0 bytes transferred to CPU RAM.

## Acceptance gate
`go test -tags vulkan -v ./internal/compute -run "TestVulkanAttentionApprox|TestVulkanPrefillBatch" -count=1` passes cleanly.

## Closure binding
Resolving commit cites this ticket and carries `(fak compute)`.

**Comment by anthony-chaudhary** [comment]
2026-09-08 local-worktree landing audit: detached 7c73f52869 was NOT landed. Current-base overlay built/vetted but failed real Strix compilation because its stale vulkan_ops.go removes VulkanKVScratchpad.ExecuteAttentionWithDequantOnce added by later #12186; vulkan_kv_test.go then fails at lines 117/169. Local tagged test selected no CGO tests, so it was rejected as a witness. Candidate and its three session artifacts remain preserved for a current-base redesign.

**Comment by anthony-chaudhary** [comment]
### Autonomous Resolution & Audit Receipt: Issue #12181

#### 1. Ground Truth Audit & Ancestry Verification
- Commit \
